WebDriverAgentLib/Routing/FBSession.m

```diff
@@ -166,7 +166,8 @@ - (void)kill
 
 - (XCUIApplication *)activeApplication
 {
-  if (nil != self.testedApplication) {
+  BOOL isAuto = [self.defaultActiveApplication isEqualToString:FBDefaultApplicationAuto];
+  if (isAuto && nil != self.testedApplication) {
```

Comment:
I would say if defaultActiveApplication is provided and is different from testedApplication then we must verify first if this application's state is `running in foreground` and return it immediately. Otherwise continue as is

Reply:
makes sense

Reply:
https://github.com/appium/WebDriverAgent/pull/862/commits/9837d335fb218612c54d5ac6190f7803169c15e6